app/models/permitted_params.rb

```diff
@@ -262,6 +236,155 @@ def wiki_content
     params.require(:content).permit(*self.class.permitted_attributes[:wiki_content])
   end
 
+  def timeline
+    acceptable_options_params = ["exist", "zoom_factor", "initial_outline_expansion", "timeframe_start",
+    "timeframe_end", "columns", "project_sort", "compare_to_relative", "compare_to_relative_unit",
+    "compare_to_absolute", "vertical_planning_elements", "exclude_own_planning_elements",
+    "planning_element_status", "planning_element_types", "planning_element_responsibles",
+    "planning_element_assignee", "exclude_reporters", "exclude_empty", "project_types",
+    "project_status", "project_responsibles", "parents", "planning_element_time_types",
+    "planning_element_time_absolute_one", "planning_element_time_absolute_two",
+    "planning_element_time_relative_one", "planning_element_time_relative_one_unit",
+    "planning_element_time_relative_two", "planning_element_time_relative_two_unit",
+    "grouping_one_enabled", "grouping_one_selection", "grouping_one_sort", "hide_other_group"]
+
+    # Options here will be empty. This is just initializing it.
+    whitelist = params.require(:timeline).permit(:name, options: {})
+
+    params['timeline']['options'].each do |key, _value|
+      whitelist['options'][key] = params['timeline']['options'][key]
+    end if params['timeline'].has_key?('options')
+
+    whitelist.permit!
+  end
+
+  def pref
+    params.require(:pref).permit(:hide_mail, :time_zone, :impaired,
+                                 :comments_sorting, :warn_on_leaving_unsaved,
+                                 :theme)
+  end
+
+  def project(instance = nil)
+    whitelist = params.require(:project).permit(:name,
+                                                :description,
+                                                :is_public,
+                                                :identifier,
+                                                :project_type_id,
+                                                custom_fields: [],
+                                                work_package_custom_field_ids: [],
+                                                type_ids: [],
+                                                enabled_module_names: [])
+
+
+    if instance && (instance.new_record? || current_user.allowed_to?(:select_project_modules, instance))
+      whitelist.permit(enabled_module_names: [])
+    end
+
+    whitelist.tap do
+      break if params[:project][:custom_field_values].nil?
+      whitelist[:custom_field_values] = params[:project][:custom_field_values]
+    end
+
+    whitelist
```

Comment:
Ok, didn't see that. What about `break whitelist`? Not sure if this is better but looks more compact.
Otherwise you don't need the `tap` at all, do you? Wouldn't an `unless` have the same meaning?